### ISSUE TITLE
Administrative Weekly Report Function

### ISSUE THREAD

**Issue by GoogleCodeExporter** [MacaroonCookie/Kickstand#53]
```
Please note that this is of low priority and should wait until after the MCV 
model, even if it would be really cool to have. I imagine that this would 
murder the current incarnation of kickstand if implemented, since even 
collecting the weekly report for one person hurts it.

Summary of enhancement (ie. what will it do):
Produce useful statistics about what has gone down and who has touched what in 
the past 7 days. A pie chart showing people's respective share of the workload 
would be amazing. For other features here, the ability to choose an arbitrary 
report length (say, monthly report, or last 4 hours report, etc.), and the 
ability to selectively remove people from the chart (perhaps an auto-generated 
list of checkboxes so we could do things like uncheck all of the staff members, 
root, and Nagios from the list to just get students, or other combos for other 
purposes) would be of high want.

Expanding upon this, we can work with the students to have a "what broke" 
report. Kickstand events already have an "Event Type" option that could be 
utilized to mark what broke in what timeframe, and then the "weekly report" 
could include this information, or could be a separate "node report" page.

What is the expected input?
1. Just visiting the page.
2. Selecting/Deselecting people
3. Changing Report Timeframe

What is the expected output?
1. Default of everybody minus root and Nagios (with check boxes to include if 
wanted)
2. Dynamically changing pie chart such that people removed or added from the 
display would be removed from or added to the pie chart, with percentages 
changing dynamically and no page refresh needed.
3. Page reloads, using new timeframe and person selection either saved or 
flipped back to default (possibly an extra option here; have "Change selection 
to default" next to the update/submit button for the timeframe change, where it 
either saves what you had or puts it back to default).

How would you prioritize this enhancement?
Right now, low. Once the MCV model is in place, moderate.

Please provide any additional information below.
Let me know if you'd like me to be more detailed about anything!
```

Original issue reported on code.google.com by `kaon...@gmail.com` on 3 Jun 2014 at 2:12
